python/nvstrings/nvstrings.py

```diff
@@ -423,6 +423,8 @@ def __getitem__(self, key):
             end = self.size() if key.stop is None else key.stop
             step = 1 if key.step is None or key.step == 0 else key.step
             # negative slicing check
+            start = -1 * self.size() if start < -1 * self.size() else start
+            end = -1 * self.size() if end < -1 * self.size() else end
```

Comment:
Should these just be modulo operations instead? It would greatly simplify this logic.

Reply:
The following scenario would yield wrong result

```
start = -1
size = -2

size % start = 0
```
But expected is `-1`

Reply:
@shwina how do we handle this normally for columns?

Reply:
@rgsl888prabhu I think this should also handle positive numbers where they're larger than self.size(), yes?

Reply:
It has been taken care in the cpp side of nvstrings.

Reply:
We use the `indices` method of the slice. Chatted briefly with @rgsl888prabhu about it